packages/now-cli/src/commands/env/add.ts

```diff
@@ -98,15 +131,59 @@ export default async function add(
 
   if (stdInput) {
     envValue = stdInput;
-  } else if (isSystemEnvVariable(envName)) {
-    envValue = '';
-  } else {
+  } else if (envType === ProjectEnvType.Plaintext) {
     const { inputValue } = await inquirer.prompt({
-      type: 'password',
+      type: 'input',
       name: 'inputValue',
       message: `What’s the value of ${envName}?`,
     });
+
     envValue = inputValue || '';
+  } else if (envType === ProjectEnvType.Secret) {
+    let secretId: string | null = null;
+
+    while (!secretId) {
+      let { secretName } = await inquirer.prompt({
+        type: 'input',
+        name: 'secretName',
+        message: `What’s the value of ${envName}?`,
+      });
+
+      secretName = secretName || '';
+
+      if (secretName[0] === '@') {
+        secretName = secretName.slice(1);
+      }
+
+      try {
+        const secret = await client.fetch<Secret>(
+          `/v2/now/secrets/${encodeURIComponent(secretName)}`
+        );
+
+        secretId = secret.uid;
+      } catch (error) {
+        if (error.status === 404) {
+          output.error(
+            `Please enter the name of an existing Secret (can be created with ${code(
```

Comment:
```suggestion
            `Please enter the name of an existing Secret (can be created with ${getCommandName(
```

Reply:
Updated in https://github.com/vercel/vercel/pull/5372/commits/0d191114447a377d4c3782c24a9e51ef095256a5.